packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -713,6 +808,89 @@ impl ScriptTransactionBuilder {
         })
     }
 
+    async fn assemble_tx(
+        self,
+        required_balances: Vec<RequiredBalance>,
+        fee_index: u16,
+        consensus_parameters: &ConsensusParameters,
+        dry_runner: impl DryRunner,
+    ) -> Result<Script> {
+        let mut tx = FuelTransaction::script(
+            0, // default value - will be overwritten
+            self.script.clone(),
+            self.script_data.clone(),
+            self.generate_fuel_policies_assemble(),
+            resolve_fuel_inputs(
+                self.inputs.clone(),
+                self.num_witnesses()?,
+                &self.unresolved_witness_indexes,
+            )?,
+            self.outputs.clone(),
+            self.witnesses.clone(),
+        );
+
+        if let Some(max_fee) = self.tx_policies.max_fee() {
```

Comment:
Same for others transaction type also : Is it necessary to set the `max_fee` here because we try to edit it again after estimation can we change the current behavior : `set -> override with assemble_tx -> maybe set again` to `set with_assemble_tx -> maybe set again` and remove this

Reply:
Good catch. Removed it here: cd6c12b9